src/side-navigation/nav-item.js

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+
+import * as React from 'react';
+import {getOverrides} from '../helpers/overrides.js';
+import {
+  StyledItemContainer,
+  StyledSubNavContainer,
+  StyledLink,
+  StyledItem,
+} from './styled-components.js';
+import type {NavItemPropsT} from './types.js';
+
+export default class NavItem extends React.Component<NavItemPropsT> {
+  static defaultProps = {
+    overrides: {},
+    onSelect: null,
+    onClick: null,
+    onKeyDown: null,
+  };
+
+  handleClick = (params: {e: Event, item: *}) => {
+    const {item, e} = params;
+    const {onSelect} = this.props;
+    if (typeof onSelect === 'function') {
+      onSelect({item});
+    }
+    this.props.onClick && this.props.onClick(e);
+  };
+
+  handleKeyDown = (params: {e: KeyboardEvent, item: *}) => {
+    const {item, e} = params;
+    if (e.key === 'Enter') {
+      const {onSelect} = this.props;
+      if (typeof onSelect === 'function') {
+        onSelect({item});
+      }
+    }
+    this.props.onKeyDown && this.props.onKeyDown(e);
+  };
+
+  render() {
+    const {
+      activePath,
+      index,
+      isActive,
+      item: topLevelItem,
+      navLevel,
+      overrides,
+      renderItem,
+    } = this.props;
+    const [ItemContainer, itemContainerProps] = getOverrides(
+      overrides.NavItemContainer,
+      StyledItemContainer,
+    );
+    const [Link, linkProps] = getOverrides(overrides.Link, StyledLink);
+    const [Item, itemProps] = getOverrides(overrides.NavItem, StyledItem);
+    const [SubNavContainer, subNavContainerProps] = getOverrides(
+      overrides.SubNavContainer,
+      StyledSubNavContainer,
+    );
+
+    const renderNavItem = (item, level, index) => {
+      const sharedProps = {
+        $active: isActive ? isActive(item, activePath) : false,
+        $level: level,
+        $selectable: !!item.path,
+      };
+      return (
+        <ItemContainer
+          key={`${index}-level${level}-${item.title}`}
+          {...sharedProps}
+          {...itemContainerProps}
+        >
+          {typeof renderItem === 'function' ? (
+            renderItem(item)
+          ) : (
+            <>
+              {item.path ? (
```

Comment:
should we decouple the routing logic from this component, and just expose a change handler? maybe something like this https://github.com/trendmicro-frontend/react-sidenav#react-router-v4-with-react-v16?

Reply:
There is no much routing logic, we may remove the default link usage in the markup tho, what do you think?

Reply:
in the current implementation, if one does not provide a path, the elements won't be rendered as clickable elements, right?

yeah, probably that would help

what do you think about that approach I linked? maybe just adding an `id` to the route elements, and let consumers handle navigation in the change handlers?

Reply:
I think the clickable part we should still handle internally. The `path` prop currently serves as an identifier, but maybe the naming is not perfect. Don't want to use `id` here as it's a valid attr. 
I'll change the link to be just a regular clickable element and let the customers handle it through the `onChange` only.

Reply:
oh yeah, that makes sense - maybe instead of the path, we should add a boolean prop like `navigatable` or something like this?

Reply:
How about `linkId` or `itemId` name?

Reply:
either sounds great to me